src/main/java/views/monthly_menu/MonthMenuV.java

```diff
@@ -0,0 +1,188 @@
+package views.monthly_menu;
+
+import entities.Category;
+import entities.Expense;
+import entities.SessionStorage;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+/**
+ * The view class for creating/updating the Month Menu view. This class implements the
+ * ActionListener interface. This class calls the controller class to get the
+ * MonthMenuOD object, and use the output to set up success and fail view.
+ */
+public class MonthMenuV implements ActionListener {
+    views.monthly_menu.UpdateViewC controller;
+    JButton addExpense;
+    JButton editExpense;
+    JButton addCategory;
+    JButton editCategory;
+    JButton generateSummary;
+
+    /**
+     * Set up success and fail view. The success view shows the Month Menu
+     * window showing the ID for the month accessed, buttons to add/edit
+     * expenses/categories, and the button to generate summary. The Month
+     * Menu window also shows the list of expenses and categories currently
+     * stored in the MonthlyStorage. The fail view shows a pop-up window
+     * showing the error message.
+     * Code inspired by https://stackoverflow.com/questions/5621338/how-to-add-jtable-in-jpanel-with-null-layout
+     * @param controller the controller class to get output data
+     * @param session the SessionStorage holding the required MonthlyStorage
+     * @param monthID the monthID of the required MonthlyStorage
+     */
+    public MonthMenuV(views.monthly_menu.UpdateViewC controller, SessionStorage session, int monthID) {
```

Comment:
I am thinking you might want to create smaller, private classes to delegate some of these tasks to, since this method is really long! for example, replacing anything that you had to comment an explanation for with a private method would work really well, especially since you'll already have half your Javadoc!

Reply:
Unfortunately, this is the only problem I cannot solve! I've tried have more smaller private methods. However, considering that we need to add code for button reactions later on, I ended up having to put some of the methods back to the constructor.

Reply:
I think if you can't avoid it, a long constructor in the view should be ok, because most of the code is just UI stuff and it is pretty well commented

Reply:
Makes sense!